tests/components/recorder/test_init.py

```diff
@@ -202,3 +203,18 @@ def test_recorder_setup_failure():
         rec.join()
 
     hass.stop()
+
+
+async def test_defaults_set(hass):
+    """Test the config defaults are set."""
+    async def mock_setup(hass, config):
```

Comment:
So if `mock_setup` never runs (for any reason, i.e. mocking issue, change in how the code works), this test will pass because it's not asserting that the method ran.  It would be better to declare a variable that is set to true within the mock setup method and then asserted after you expect it to run.

Reply:
Done!

Reply:
It is always great to see someone also review the tests 👍